### PR TITLE
add code of conduct

### PR DESCRIPTION
**Pull request by kain88-de**
This adds the Code of Conduct  as discussed on the [mailing list](https://groups.google.com/forum/#!topic/mdnalysis-devel/ueAJv47RftM). 

This is still work in progress and we should take some time to review it. Github should make it easier to add comments.

### CODE REVIEW

File: package/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+4. **Be respectful**. Not all of us will agree all the time, but disagreement is
+   no excuse for poor behavior or poor manners. We might all experience some
+   frustration now and then, but we cannot allow that frustration to turn into a
+   personal attack. It’s important to remember that a community where people
+   feel uncomfortable or threatened is not a productive one.
+
+5. **Be careful in the words that you choose**. Be kind to others. Do not insult
+   or put down other community members. Harassment and other exclusionary
+   behavior are not acceptable. This includes, but is not limited to:
+   * Violent threats or violent language directed against another person
```
Comment (jbarnoud) [inline package/CODE_OF_CONDUCT.md]: A threat does not have to be violent to be unacceptable. I would just say "Threats or violent language...".

File: package/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+at [*User-mailing-list*](mailto:mdnalysis-discussions@googlegroups.com).
+
+
+1. **Be friendly and patient**.
+
+2. **Be welcoming**. We strive to be a community that welcomes and supports
+   people of all backgrounds and identities. This includes, but is not limited
+   to, members of any race, ethnicity, culture, national origin, color,
+   immigration status, social and economic class, educational level, sex, sexual
+   orientation, gender identity and expression, age, physical appearance, family
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: Should we add "political views"?

File: package/CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+This code applies equally to founders, developers, mentors and new
+community members, in all spaces managed by MDAnalysis. This
+includes the mailing lists, our GitHub organizations, our chat rooms,
+in-person events, and any other forums created by the project team. In
+addition, violations of this code outside these spaces may affect a
+person's ability to participate within them.
+
+If you believe someone is violating the code of conduct, we ask that you report
+it by
+emailing [*User-mailling-list*](mailto:mdnalysis-discussions@googlegroups.com)
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: spelling: User-mailing-list

File: package/CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+or in anonymous by filling out [*this form*](TODO).
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: Any idea how to do anonymous forms?

File: package/CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+4. **Be respectful**. Not all of us will agree all the time, but disagreement is
+   no excuse for poor behavior or poor manners. We might all experience some
+   frustration now and then, but we cannot allow that frustration to turn into a
+   personal attack. It’s important to remember that a community where people
+   feel uncomfortable or threatened is not a productive one.
+
+5. **Be careful in the words that you choose**. Be kind to others. Do not insult
+   or put down other community members. Harassment and other exclusionary
+   behavior are not acceptable. This includes, but is not limited to:
+   * Threats or violent language directed against another person
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: all bullet points should start with lower case

File: package/CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+
+## Reporting
+
+If someone makes you or any other contributor feel unsafe or unwelcome,
+please report this in a timely manner. Code of conduct violations reduce
+the value of the community for everyone and we take them seriously.
+
+You can file a report by emailing
+[*special-list?*](TODO) or by filing out
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: To be consistent with the above, this is user-mailing-list.

File: package/CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+[this form](TODO). For more details or
+information on reporting in-person at an event, please see our Reporting
+Guidelines.
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: We don't have Reporting Guidelines so remove for now?

File: package/CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+The online form gives you the option to keep your report anonymous or request
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: I think, for anonymous reports it would be important that some tangible evidence is provided. If it's anonymous and only actions/observations are reported then it might be very hard to do anything while at the same time there's a possibility for creating a bad atmosphere. Although, I admit, I have no experience with anonymous reporting so perhaps it works well.

File: package/CODE_OF_CONDUCT.md (hunk 2 above)
Comment (kain88-de) [inline package/CODE_OF_CONDUCT.md]: There stands political belief later in the list

File: package/CODE_OF_CONDUCT.md (hunk 8 above)
Comment (kain88-de) [inline package/CODE_OF_CONDUCT.md]: I also have no idea. I would be OK though if we add a non-public mailing list that people can use to write emails to. I would like as least one non-public option. That is important for me. The form might be nice since it lowers the barrier. 

Here is the jupyter [form](https://docs.google.com/forms/d/e/1FAIpQLSfUuaGbXrH0anroNIkbc-dBrn-ISsC_BA57IRt57aQW85nK6w/viewform?c=0&w=1)

**Comment by kain88-de** [comment]
@orbeckst I would like to have an approval of all @MDAnalysis/coredevs on this PR before we merge this. 

I updated the text a good bit with changes from jupyter and @orbeckst suggestions.

**Comment by jbarnoud** [comment]
Is there a reason for the code of conduct to be in `package` rather than in the root of the repo?

**Comment by kain88-de** [comment]
There isn't any particular reason. I just thought it would fit better there since most community related documents are there.

**Comment by richardjgowers** [comment]
@kain88-de thanks for this, it looks good. Maybe @jbarnoud is right and we should move it to the base of the repo so it's visible on the homepage.

**Comment by kain88-de** [comment]
We can link to it on the homepage and the mailing lists after this PR is merged. Otherwise I don't have a special opinion to where this file should be in the end. 

But a important questions is. Should we supply an anonymous google doc form or a mdnalysis-coc@... mailing list that isn't public? I would prefer the mailing  list option. I would like to know what others think.

Information about creating forms is [here](https://www.google.com/forms/about/)

**Comment by jbarnoud** [comment]
As a general rule, I prefer email. I wonder what about of garbage we should expect with a mailing list that can be addressed by anyone without registration (registration creates a too high barrier).

**Comment by mnmelo** [comment]
I'm a bit divided between form vs. private mailing list (the latter is handier, but it is not anonymous and is susceptible to spamming). Since the opinion here is tending to the mailing list I'd say we roll with that, and adapt if we see it doesn't work.

Otherwise I think the text is great.

**Comment by orbeckst** [comment]
> On 19 Jan, 2017, at 15:55, Max Linke <notifications@github.com> wrote:
> 
> I also have no idea. I would be OK though if we add a non-public mailing list that people can use to write emails to. I would like as least one non-public option. That is important for me.

Yes, definitely. I myself actually would not want to report to the users or devs list.

I can set up a group like mdnalysis-business. What should be the name? mdnalysis-incidents ? 

> The form might be nice since it lowers the barrier.
> 
> Here is the jupyter form

Oh, it's just a Google Form. That's trivial to set up. I can create one. The main question is who has access to the form data – it cannot be a single person because what if the complaint is about that person? I don't know if there's an easy way to share access to a form because it might be tied to a google account. Any ideas?

**Comment by orbeckst** [comment]
Both form and list are easy to set up.

We'll monitor the spam on the list. We'll put down a bunch of coredevs as moderators and whoever sees an email first flags it as spam or lets it through.

If it becomes unsustainable then we will just keep the form.

> On 20 Jan, 2017, at 07:14, mnmelo <notifications@github.com> wrote:
> 
> I'm a bit divided between form vs. private mailing list (the latter is handier, but it is not anonymous and is susceptible to spamming). Since the opinion here is tending to the mailing list I'd say we roll with that, and adapt if we see it doesn't work.

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by kain88-de** [comment]
OK so the consensus seems to be for just a non-public mailing list. I would be for `mdnalysis-coc@...`

**Comment by orbeckst** [comment]
I feel mdnalysis-coc@ is not very clear … "coc"? abuse or incidents or complaints or anything along those  lines would be clearer in my opinion.

I would still also offer the anonymous form because (a) easy to do and (b) might lower barrier. I would just add that it is more difficult to resolve any situation if there is no way to follow up.

By the way, what is the procedure going to be if/when we receive an incident report? Will the coredevs discuss this on the "coc" list or on the business list? What further steps will in general be taken and who takes them?

Basically, if there is no follow-up then complaints are pointless.



> On 20 Jan, 2017, at 10:22, Max Linke <notifications@github.com> wrote:
> 
> OK so the consensus seems to be for just a non-public mailing list. I would be for mdnalysis-coc@...
> 
> 

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by kain88-de** [comment]
About the procedure when a report occurs. We can have a look at the jupyter procedure. My intuitive approach would be to talk to all involved and deescalate the situation. A second step is to close discussion and tell everyone to cool off for some time.

**Comment by kain88-de** [comment]
@orbeckst How do you like mdnalysis-conduct@... ?

**Comment by kain88-de** [comment]
I found some more information about code of conducts and their enforcement [here](http://contributor-covenant.org/)

**Comment by kain88-de** [comment]
[This talk](http://www.slideshare.net/aeschright/enforcing-your-code-of-conduct-effective-incident-response) actually gives some good rules how we should proceed if their ever is a violation.

**Comment by orbeckst** [comment]
I like it!
> On 21 Jan, 2017, at 10:53, Max Linke <notifications@github.com> wrote:
> 
> @orbeckst How do you like mdnalysis-conduct@... ?

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by orbeckst** [comment]
I created *[mdnalysis-conduct](https://groups.google.com/forum/#!forum/mdnalysis-conduct)* as a group 
* public posting allowed
* only members can view
* membership only by invitation
* created as a "[collaborative inbox](https://support.google.com/a/answer/167430?hl=en)", which means that there features available such as assigning topics (posts) to people... maybe it's useful

@kain88-de : added you as an owner (with your gmail address); you can start re-organizing and inviting people. 

I tried sending a test email from another account and that seems to have worked (the first email is now a "test" email). I also checked that the group's contents are not visible if I am just browsing https://groups.google.com/forum/#!forum/mdnalysis-conduct without being logged in as a group member.

The welcome message and the group summary still needs refining/updating with links to CoC etc.

**Comment by orbeckst** [comment]
As part of the links at http://contributor-covenant.org/ (from https://github.com/MDAnalysis/mdanalysis/pull/1165#issuecomment-274277638) I found the [Django CoC Enforcement Manual](https://www.djangoproject.com/conduct/enforcement-manual/) well written and fairly applicable and I think we could adopt it.

The slides [Enforcing Your Code of Conduct: effective incident response ](http://www.slideshare.net/aeschright/enforcing-your-code-of-conduct-effective-incident-response) are also interesting reading. Basically, be prepared *before* anything happens and understand that a CoC is not really worth a lot if you are not willing (or prepared) to enforce it.

File: package/CODE_OF_CONDUCT.md (hunk 2 above)
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: ok

**Comment by kain88-de** [comment]
@orbeckst I added a short enforcement section. Can you have a look over it. I tried to set a general timeframe for reactions but the text should also give us some freedom to take more/less time depending on our personal free time we have when an incident occurs.

File: package/CODE_OF_CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+confidentiality with regard to the reporter of an incident.
+
+The online form gives you the option to keep your report anonymous or request
+that we follow up with you directly. While we cannot follow up on an anonymous
+report, we will take appropriate action.
+
+# Enforcement
+
+When a report is send to us we will reply a soon as possible to confirm receipt,
+we strive to answer in less then 24 hours. We will review the incident and
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: then --> than

File: package/CODE_OF_CONDUCT.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+- what happened
+- whether this event constitutes a code of conduct violations
+- who, if anyone, was the bad actor
+- whether this is an ongoing situation
+
+This information will collected in writing. We strive to reach a resolution in a
+week after conformation. Once we agree on a resolution, but before it is
+enacted, we will contact the original reporter and any other affected parties
+and explain the proposed resolution. We will ask if this resolution is
+acceptable and note feedback for the record. However we are not required to act
```
Comment (orbeckst) [inline package/CODE_OF_CONDUCT.md]: However, (comma)

**Comment by orbeckst** [comment]
I created an anonymous Google doc form: https://goo.gl/forms/w2IwBKkY3oT0aVEB3 – I used our CoC intro and added a bit at the end. In particular we need to have the CoC also on our webpage at http://mdanalysis.org/pages/conduct/ (MDAnalysis/MDAnalysis.github.io#34).

I made @kain88-de the owner of the form, I am currently an editor, but for right now @kain88-de should manage it, including adding other editors once we have figured out who should do the CoC work.

It was based on Jupyter's form https://docs.google.com/forms/d/e/1FAIpQLSfUuaGbXrH0anroNIkbc-dBrn-ISsC_BA57IRt57aQW85nK6w/viewform?c=0&w=1

File: package/CODE_OF_CONDUCT.md [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+When a report is send to us we will reply a soon as possible to confirm receipt,
+we strive to answer in less then 24 hours. We will review the incident and
+determine, to the best of our ability
+
+- what happened
+- whether this event constitutes a code of conduct violations
+- who, if anyone, was the bad actor
+- whether this is an ongoing situation
+
+This information will collected in writing. We strive to reach a resolution in a
```
Comment (jbarnoud) [inline package/CODE_OF_CONDUCT.md]: information will *be* collected

**Comment by jbarnoud** [comment]
Out of curiosity, what would be the procedure to change something in the code of conduct once enacted?

I remember that a french python group had an issue recently with its code of conduct. It was saying that the group would not officially promote any discriminatory event; this caused a big fuss when the board of the group refused to promote a django girls event on the webpage because men were not allowed to enrol. The discussion on whether or not the group had to clarify the code of conduct to allow the official promotion of such event was heated. All of this to say that what could look obvious now may have unexpected implications latter, and that we may be lead to amend the code of conduct at some point. I not super clear before coffee.

**Comment by orbeckst** [comment]
> On 28 Jan, 2017, at 00:38, Jonathan Barnoud <notifications@github.com> wrote:
> 
> Out of curiosity, what would be the procedure to change something in the code of conduct once enacted?

Good question. Basically you're asking what kind of governance structure we have. We haven't formalized anything, maybe we should. In practice the core-devs informally vote on whatever appears to be relevant for the project. Although I suppose that simple majority is sufficient, I don't remember a situation where a decision was made against any opposing views – maybe my memory isn't great but I'd like to think that we discussed anything contentious sufficiently so that in the end everyone was on board. Might not always work like this in the future, of course.

For CoC changes I would suggest voting of core-devs.

Btw, we also need ~3 people who will take care of CoC Reports. These positions can rotate, of course.

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by kain88-de** [comment]
@jbarnoud I would say the core-devs have to vote on a change of the COC, similar like they should all agree to adopting this coc.  

@orbeckst do you want to have an explicit coc board? My thought was all owners should be equally reponsible for the coc.

**Comment by orbeckst** [comment]
> @orbeckst do you want to have an explicit coc board? My thought was all owners should be equally reponsible for the coc.

That's fine with me – less complicated. I suggested a group of ~3 from my experience with committees: if there are too many people then no-one feels responsible to do anything. But we should certainly just try out "whole board" – we're pretty good at communicating with each other and if we find that the process does not work well then we can think of more admin structures.

**Comment by orbeckst** [comment]
Anything else to add here? I would be ready to merge it.

File: package/CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+filing out [this form][conduct-form]. The project team is obligated to maintain
+confidentiality with regard to the reporter of an incident.
+
+The online form gives you the option to keep your report anonymous or request
+that we follow up with you directly. While we cannot follow up on an anonymous
+report, we will take appropriate action.
+
+# Enforcement
+
+When a report is send to us we will reply a soon as possible to confirm receipt,
```
Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "is sent"

Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "as soon as"

Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "receipt;" (semi-colon)

File: package/CODE_OF_CONDUCT.md [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+we strive to answer in less than 24 hours. We will review the incident and
+determine, to the best of our ability
+
+- what happened
+- whether this event constitutes a code of conduct violations
```
Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "violation"

File: package/CODE_OF_CONDUCT.md [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+- who, if anyone, was the bad actor
```
Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "was at fault" instead of "was the bad actor"

File: package/CODE_OF_CONDUCT.md [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+- whether this is an ongoing situation
+
+This information will be collected in writing. We strive to reach a resolution
+in a week after conformation. Once we agree on a resolution, but before it is
```
Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "within a week of confirmation. Once a resolution has been agreed upon,"

File: package/CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+enacted, we will contact the original reporter and any other affected parties
+and explain the proposed resolution. We will ask if this resolution is
+acceptable and note feedback for the record, however we are not required to act
```
Comment (mnmelo) [inline package/CODE_OF_CONDUCT.md]: "for the record. We are, however, not required"

**Comment by mnmelo** [comment]
Sorry for the scattered remarks on the Enforcement text. I'm in a rush and can't make a proper PR with all of it.

**Comment by kain88-de** [comment]
> Sorry for the scattered remarks on the Enforcement text. I'm in a rush and can't make a proper PR with all of it.

just having remarks is fine. I can later add these changes. Thanks for taking the time

File: CODE_OF_CONDUCT.md [hunk 17]
```diff
@@ -0,0 +1,159 @@
...
+we strive to answer in less than 24 hours. We will review the incident and
+determine, to the best of our ability
+
+- what happened
+- whether this event constitutes a code of conduct violation
+- who, if anyone, was at fault
+- whether this is an ongoing situation
+
+This information will be collected in writing. We strive to reach a resolution
+within a week after conformation. Once a resolution has been agreed upon, but before it is
```
Comment (mnmelo) [inline CODE_OF_CONDUCT.md]: Still needs fixing:
"within a week *of* *confirmation*" or do you really mean "conformation"?
(Also note the "of" instead of "after")

**Comment by orbeckst** [comment]
@MDAnalysis/coredevs please read and approve. It is important that everyone within a leading role in the MDAnalysis org

1. has read the CoC and
2. is ready to stand behind it 100% and
3.  is ready to enforce it
4. has confirmed (by approving the PR) the above --- sort of your personal signature.

**Comment by orbeckst** [comment]
@MDAnalysis/coredevs , I added everyone as a reviewer (except @kain88-de). I would expect that anyone interested in the overall project would be able to review this important policy document over the next two days.

Thank you.

Oliver

**Comment by kain88-de** [comment]
The rOpenSci community recently collected [some thoughts](https://ropensci.org/blog/blog/2016/12/21/commcallv12-review-coc?utm_source=NumFOCUS+Master+List&utm_campaign=77f584f624-EMAIL_CAMPAIGN_2017_01_19&utm_medium=email&utm_term=0_d97ca30fde-77f584f624-513584257&mc_cid=77f584f624&mc_eid=332d28359c) on creating a CoC. Worth a short read also with regard to enforcement. Our current process seems to be pretty much along the lines they recommend.

**Comment by jandom** [comment]
Looks good :+1:

**Comment by kain88-de** [comment]
@jandom could you please hit the approve button at the top of this PR as well. Thank you

**Comment by tylerjereddy** [comment]
Approved / signed.

**Comment by kain88-de** [comment]
@orbeckst can you update the descriptions in the google group to link to the CoC. I think then we are done.

**Comment by orbeckst** [comment]
Yes, will do that. I am also composing a general email to the mailing lists.

- [ ] update mdnalysis-conduct header
- [ ] update report form header
- [ ] send announcement to lists
